penrose-web/src/compiler/Style.ts

```diff
@@ -1,7 +1,2131 @@
+
+  const fieldDict = trn[nm];
+
+  if (!fieldDict) {
+    // TODO (errors / warnings): Should this be fatal?
+    const err = `Err: Sub obj '${nm}' has no fields; can't delete path '${pathStr}'`;
+    return addWarn(trans, err);
+  }
+
+  const prop: FieldExpr<VarAD> = fieldDict[fld];
+
+  if (!prop) {
+    // TODO (errors / warnings): Should this be fatal?
+    const err = `Err: Sub obj '${nm}' already lacks field '${fld}'; can't delete path ${pathStr}`;
+    return addWarn(trans, err);
+  }
+
+  if (prop.tag === "FExpr") {
+    // Deal with GPI aliasing (i.e. only happens if a GPI is aliased to another, and some operation is performed on the aliased GPI's property, it happens to the original)
+    // COMBAK: should path aliasing have destructive effects on the translation (e.g. add or delete)? maybe it should only happen in lookup? Deleting an aliased path should just delete the alias, not its referent?
+    // TODO: Test this
+
+    if (prop.contents.tag === "OptEval") {
+      if (prop.contents.contents.tag === "FieldPath") {
+        const p = prop.contents.contents;
+        if (varsEq(p.name.contents, name.contents) && varsEq(p.field, field)) {
+          // TODO (error)
+          const warn = `path was aliased to itself`;
+          return addWarn(trans, warn);
+        }
+        return deleteProperty(trans, p.name, p.field, property);
+      }
+    }
+
+    // TODO (error)
+    const warn = `Err: Sub obj '${name.contents.value}' does not have GPI '${field.value}'; cannot delete property '${property.value}'`;
+    return addWarn(trans, warn);
+
+  } else if (prop.tag === "FGPI") {
+    // TODO(error, warning): check if the property is member of properties of GPI
+    const gpiDict = prop.contents[1];
+    delete gpiDict.prp;
+    return trans;
+  } else throw Error("unknown tag");
+};
+
+// Note this mutates the translation, and we return the translation reference just as a courtesy
+const deleteField = (trans: Translation, name: BindingForm, field: Identifier): Translation => {
+  // TODO (errors)
+  let trn = trans.trMap;
+  const fieldDict = trn[name.contents.value];
+
+  if (!fieldDict) {
+    // TODO(errors / warnings)
+    const warn = `Err: Sub obj '${name.contents.value}' has no fields; can't delete field '${field.value}'`;
+    return addWarn(trans, warn);
+  }
+
+  if (!(field.value in fieldDict)) {
+    // TODO(errors / warnings)
+    const warn = `Warn: SubObj '${name.contents.value}' already lacks field '${field.value}'`;
+    return addWarn(trans, warn);
+  }
+
+  delete fieldDict[field.value];
+  return trans;
+};
+
+// NOTE: This function mutates the translation
+// rule Line-delete
+const deletePath = (trans: Translation, path: Path): Either<StyErrors, Translation> => {
+  // TODO(errors): Maybe fix how warnings are reported? Right now they are put into the "translation"
+  if (path.tag === "FieldPath") {
+    const transWithWarnings = deleteField(trans, path.name, path.field);
+    return Right(transWithWarnings);
```

Comment:
Minor: why `let`? Looks like `const` is better suited for this. BTW these are all caught by ESLint. You can start using it to spot lower-level issues like this.